### PR TITLE
Faster cuda worklist

### PR DESCRIPTION
**Pull request by thorulf4**

### CODE REVIEW

File: src/cuda/cuda_worklist/analysis.cu [hunk 1]
```diff
@@ -141,8 +142,12 @@ void cuda_worklist::execute_analysis(Node* nodes, int node_count, Transfer* tran
     dev_transfers = cuda_allocate_memory<Transfer>(sizeof(Transfer)*transfer_count);
     cuda_copy_to_device(dev_transfers, transfers, sizeof(Transfer)*transfer_count);
   
+    Stopwatch lfp_watch;
+    lfp_watch.start();
     int i = 0;
+    int j = 0;
```
Comment (ZcortezZ) [inline src/cuda/cuda_worklist/analysis.cu:148]: unused j

File: src/cuda/cuda_worklist/analysis.cu [hunk 2]
```diff
@@ -141,8 +142,12 @@ void cuda_worklist::execute_analysis(Node* nodes, int node_count, Transfer* tran
     dev_transfers = cuda_allocate_memory<Transfer>(sizeof(Transfer)*transfer_count);
     cuda_copy_to_device(dev_transfers, transfers, sizeof(Transfer)*transfer_count);
   
+    Stopwatch lfp_watch;
+    lfp_watch.start();
```
Comment (NicklasHugoy) [inline src/cuda/cuda_worklist/analysis.cu:146]:
```suggestion
    Stopwatch lfp_watch;
```

Comment (NicklasHugoy) [reply src/cuda/cuda_worklist/analysis.cu:146]: Stopwatch starts when initialized

File: src/main.cpp [hunk 3]
```diff
@@ -67,7 +66,8 @@ void bit_cuda_analysis(ScTransformer& program){
         print_digraph_subgraph(nodes, std::cout, print_result, "main");
 }
 
-void bit_cuda_worklist_analysis(ScTransformer& program){
+std::vector<StatefulNode<std::set<std::string>>> bit_cuda_worklist_analysis(ScTransformer& program){
+    cuda_worklist::init_gpu();
```
Comment (ZcortezZ) [inline src/main.cpp:70]: should also add init gpu for bit_cuda

File: src/cuda/cuda_worklist/analysis.cu [hunk 4]
```diff
@@ -141,8 +142,12 @@ void cuda_worklist::execute_analysis(Node* nodes, int node_count, Transfer* tran
...
     int i = 0;
+    int j = 0;
     while(work_to_do){
+        j++;
```
Comment (NicklasHugoy) [inline src/cuda/cuda_worklist/analysis.cu:150]: `j` is not used?

**Review by ZcortezZ — CHANGES_REQUESTED** [review]

**Review by NicklasHugoy — CHANGES_REQUESTED** [review]

**Review by ZcortezZ — APPROVED** [review]

**Review by NicklasHugoy — APPROVED** [review]

**Pull request merged**